scripts/globals/dark_ixion.lua

```diff
@@ -0,0 +1,471 @@
+            -443.7, -011.5,  056.8,
+            -418.7, -013.0,  175.0,
+            -387.5, -013.8,  219.5,
+            -299.0, -010.5,  220.8,
+            -261.0, -005.5,  233.7,
+            -254.5, -005.0,  293.8,
+            -185.5,  002.5,  294.0,
+            -086.0,  000.0,  257.5,
+            -099.5,  002.0,  164.5,
+            -049.5,  002.5,  103.0,
+            0000.0,  000.0,  046.0,
+        }
+    },
+    [xi.zone.FORT_KARUGO_NARUGO_S] =
+    {
+        mobID = 17170649,
+        pathList =
+        {
+            -088.8, -068.0, -269.5,
+            -025.0, -067.0, -282.5,
+            0023.7, -068.0, -232.0,
+            0054.7, -069.7, -178.0,
+            0085.0, -064.0, -165.5,
+            0105.5, -065.0, -180.0,
+            0136.0, -059.0, -215.5,
+            0180.7, -056.0, -195.5,
+            0198.0, -057.8, -146.7,
+            0199.0, -043.0, -062.5,
+            0200.8, -040.0, -026.4,
+            0205.7, -028.0,  021.3,
+            0256.6, -022.0,  025.0,
+            0281.6, -026.5,  010.0,
+            0259.5, -029.8, -049.5,
+            0239.0, -040.5, -100.0,
+            0205.7, -057.0, -145.0,
+            0194.0, -054.0, -186.0,
+            0180.7, -047.5, -227.2,
+            0135.0, -050.8, -251.0,
+            0107.8, -061.7, -295.5,
+            0097.0, -061.5, -307.5,
+            0041.0, -068.0, -321.0,
+            -007.0, -067.9, -301.0,
+            -052.5, -067.7, -275.0,
+        }
+    },
+    [xi.zone.EAST_RONFAURE_S] =
+    {
+        mobID = 17109367,
+        pathList =
+        {
+            236.0, -020.0, -323.0,
+            245.6, -019.5, -290.0,
+            291.0, -016.0, -257.5,
+            358.5, -016.0, -259.0,
+            384.7, -015.0, -227.0,
+            405.0, -016.7, -220.0,
+            434.0, -016.7, -220.0,
+            459.0, -015.7, -225.5,
+            469.0, -015.0, -256.0,
+            496.0, -015.0, -266.6,
+            498.0, -016.0, -303.0,
+            501.0, -015.0, -327.5,
+            509.0, -005.0, -376.8,
+            537.0, -005.5, -387.0,
+            537.0, -006.0, -437.0,
+            514.0, -009.0, -450.5,
+            480.0, -009.6, -446.6,
+            438.0, -010.0, -440.5,
+            411.4, -010.0, -393.0,
+            368.5, -010.0, -407.0,
+            346.0, -007.0, -429.8,
+            318.2,  000.0, -437.6,
+            271.5,  004.3, -461.6,
+            219.7, -017.0, -334.0,
+        }
+    },
+}
+
+xi.darkixion.repop = function(mob)
+    DespawnMob(mob:getID())
+    local keys = {}
+    for k, _ in pairs(xi.darkixion.zoneinfo) do
+        table.insert(keys, k)
+    end
+
+    local randZoneID = keys[math.random(#keys)]
+    SetServerVariable('DarkIxion_ZoneID', randZoneID)
+    -- wiki says 'It can pop there in less than 10 seconds or take the whole 15 minutes'
+    SetServerVariable('DarkIxion_PopTime', os.time() + math.random(1, 15 * 60)) -- based on onGameHour function timing
+end
+
+xi.darkixion.roamingMods = function(mob)
```

Comment:
This could be more descriptive, it looks like its doing a lot more than just adding mods. Maybe break out a bunch of the stuff at the end into a reset/reinit function?

Reply:
I'll add a comment, I made it a function so I didn't have to replicate the logic in two different mob functions. Basically contains all the things that are needed when DI is roaming.